ArduCopter/mode_turtle.cpp

```diff
@@ -0,0 +1,138 @@
+#include "Copter.h"
+
+#if MODE_TURTLE_ENABLED == ENABLED
+
+#define CRASH_FLIP_EXPO 35.0f
+#define CRASH_FLIP_STICK_MINF 0.15f
+#define power3(x) ((x)*(x)*(x))
+
+// throw_init - initialise turtle mode controller
+bool ModeTurtle::init(bool ignore_checks)
+{
+    // do not enter the mode when already armed or when flying
+    if (motors->armed() || SRV_Channels::get_dshot_esc_type() == 0) {
+        return false;
+    }
+    // reverse the motors
+    hal.rcout->disable_channel_mask_updates();
+    change_motor_direction(true);
+
+    // arm
+    motors->armed(true);
+    hal.util->set_soft_armed(true);
+
+    return true;
+}
+
+void ModeTurtle::exit()
+{
+    // disarm
+    motors->armed(false);
+    hal.util->set_soft_armed(false);
+
```

Comment:
Same as above. Arming and disarming from mode entry and exit is not a good idea IMHO

Reply:
Arming checks added